### PR TITLE
Add doc for table-scan-node-partitioning-min-bucket-to-task-ratio

### PR DESCRIPTION
**Pull request by sopel39**
to be landed after: https://github.com/trinodb/trino/pull/7257

### CODE REVIEW

File: docs/src/main/sphinx/admin/properties-optimizer.rst [hunk 1]
```diff
@@ -116,3 +116,14 @@ the maximum number of joins that can be reordered at once.
...
 join output rows can be skipped.
+
+``optimizer.table-scan-node-partitioning-min-bucket-to-task-ratio``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** ``double``
+* **Default value:** ``0.5``
+
+Specifies minimal bucket to task ratio that has to be exceeded in order
+to use table scan node partitioning. When there is not enough buckets
```
Comment (raunaqmorarka) [inline docs/src/main/sphinx/admin/properties-optimizer.rst:127]: We should explain what is "table scan node partitioning"

Comment (raunaqmorarka) [inline docs/src/main/sphinx/admin/properties-optimizer.rst:127]: When there is not enough buckets... -> When the number of buckets is small compared to the number of table scan tasks then the table scan will be ...

File: docs/src/main/sphinx/admin/properties-optimizer.rst [hunk 2]
```diff
@@ -116,3 +116,14 @@ the maximum number of joins that can be reordered at once.
...
+
+``optimizer.table-scan-node-partitioning-min-bucket-to-task-ratio``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** ``double``
+* **Default value:** ``0.5``
+
+Specifies minimal bucket to task ratio that has to be matched or exceeded in order
+to use table scan node partitioning. When the table bucket count is small
+compared to number of nodes then the table scan will be distributed
```
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-optimizer.rst:128]: to the number of workers, then the table scan is distributed across all workers for improved parallelism.

**Review by mosabua — CHANGES_REQUESTED** [review]
Please update grammar and usage of "workers" as requested

**Review by mosabua — APPROVED** [review]

**Review by electrum — APPROVED** [review]

**Pull request merged**